app/transactions/admin/missions/update_transaction.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Admin
+  module Missions
+    # update mission and update recurrently if the recurrent change params is given
+    class UpdateTransaction
+      include Dry::Transaction
+
+      around :rollback_if_failure
+
+      step :update_mission
+      tee :get_updatable_missions
+      step :update_recurrently_other_missions
+
+      private
+
+      def rollback_if_failure(input, &block)
+        result = nil
+
+        Mission.transaction do
+          result = block.call(Success(input))
+          raise ActiveRecord::Rollback if result.failure?
+
+          result
+        end
+        result
+      end
+
+      def update_mission(input, mission:)
+        if mission.update(input[:params])
+          Success(input)
+        else
+          failure_message = I18n.t('activerecord.errors.messages.update_fail')
+          Failure(failure_message)
+        end
+      end
+
+      def get_updatable_missions(input, old_mission:)
+        return Success(input) unless input[:params][:recurrent_change]
+
+        input.merge!({ missions: updatable_missions(old_mission) })
+        Success(input)
+      end
+
+      def update_recurrently_other_missions(input)
+        return Success(input) unless input[:params][:recurrent_change]
+
+        failure_message = I18n.t('activerecord.errors.messages.update_fail')
+
+        input[:missions].each do |mission|
+          return Failure(failure_message) unless mission.update(input[:params])
+        end
```

Comment:
If I understand correctly:
* lets say there are 3 missions on Monday 7, Monday 14, Monday 21 (of January)
* we update Monday 7 to have a new title

=> Since we are using Monday7's `params`, Monday14 and Monday21 will also see their date change to January 7th. I'm pretty sure that's not what we want

The associated ticket is very vague and needs to be much more specific. Let's take some time to re-spec this feature